### ISSUE TITLE
Testing Framework

### ISSUE THREAD

**Issue by 138paulmiller** [GentenStudios/quartz-engine#136]
This will likely become necessary as this project grows. See this branch [here](https://github.com/138paulmiller/quartz-engine/tree/feat-unittests/QuartzTest) for a working example.

**Comment by beeperdeeper089** [comment]
Yeah, a testing framework would be pretty useful, we can work on getting that properly implemented and writing unit tests for everything once that's ready.

Once it's ready, we can probably plumb that into Travis and make it do the tests for it PR and so on.

**Comment by tobyplowy** [comment]
@138paulmiller Are you still interested on working on this?
